### PR TITLE
weather_slider_1_minute_step

### PR DESCRIPTION
**Pull request by nnngrach**
[issue](https://github.com/osmandapp/OsmAnd-iOS/issues/3779#issuecomment-2283478057)

https://github.com/user-attachments/assets/4c6488f5-c875-42e1-a774-981cd340ce02

### CODE REVIEW

**Comment by nnngrach** [comment]
In android 2.5 minutes per step.
Screenshot 2024-08-13 at 17 22 23

But it's rounding on setting.
0.0  2.5  5.0  7.5  10.0  12.5  15.0 - clear minutes from slider
0.0  3.0  5.0  8.0  10.0  13.0  15.0 - rounded minutes

Screenshot 2024-08-13 at 13 00 14
Screenshot 2024-08-13 at 13 00 25
Screenshot 2024-08-13 at 13 00 44
Screenshot 2024-08-13 at 13 01 35
Screenshot 2024-08-13 at 13 01 59

---

**Comment by nnngrach** [comment]
Test slider after refactoring.

https://github.com/user-attachments/assets/69cdd60c-9c12-4c6c-be62-308f99561bad

Also found and fixed bug with pink slider marks on mac os

Screenshot 2024-08-13 at 22 01 18 Screenshot 2024-08-13 at 23 18 03

File: Sources/Views/OAWeatherTimeSegmentedSlider.h [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+//
+//  Created by Max Kojin on 13/08/24.
+//  Copyright © 2024 OsmAnd. All rights reserved.
+//
+
+#import "OASegmentedSlider.h"
+
+static const double kForecastStepDuration = 2.5; // 2.5 minutes step
+static const int kForecastStepsPerHour = 60.0 / kForecastStepDuration; // 24 steps in hour
+static const int kForecastWholeDayMarksCount = 24 * kForecastStepsPerHour + 1; // 577 steps in slider
```
Comment (tigrim) [inline Sources/Views/OAWeatherTimeSegmentedSlider.h:13]: to  .m file

File: Sources/Views/OAWeatherTimeSegmentedSlider.mm [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+                [timeValuesTotal addObject:nextStepDate];
+            }
+        }
+        
+    }
+    // [21:00, 21:03, 21:05, 21:08, 21:10, 21:13  ...  23:58, 00:00 ...  20:58, 21:00]
+    _timeValues = timeValuesTotal;
+}
+
+
```
Comment (tigrim) [inline Sources/Views/OAWeatherTimeSegmentedSlider.mm:82]: remove empty space

File: Sources/Views/OAWeatherTimeSegmentedSlider.mm [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+{
+    return _timeValues.count;
+}
+
+- (NSDate *)getSelectedDate
+{
+    return _timeValues[[self getIndexForOptionStepsAmountWithoutDrawMark]];
+}
+
+
```
Comment (tigrim) [inline Sources/Views/OAWeatherTimeSegmentedSlider.mm:99]: remove

File: Sources/Views/OAWeatherTimeSegmentedSlider.mm [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+    NSDate *date = [OAUtilities getCurrentTimezoneDate:[NSDate date]];
+    NSInteger minimumForCurrentMark = [_currentTimezoneCalendar startOfDayForDate:date].timeIntervalSince1970;
+    NSInteger currentValue = date.timeIntervalSince1970;
+    self.currentMarkX = index == 0 ? (currentValue - minimumForCurrentMark) : -1;
+    date = [_currentTimezoneCalendar startOfDayForDate:date];
+    date = [_currentTimezoneCalendar dateByAddingUnit:NSCalendarUnitDay value:1 toDate:date options:0];
+    self.maximumForCurrentMark = [_currentTimezoneCalendar startOfDayForDate:date].timeIntervalSince1970 - minimumForCurrentMark;
+}
+
+
```
Comment (tigrim) [inline Sources/Views/OAWeatherTimeSegmentedSlider.mm:137]: remove

**Review by tigrim — APPROVED** [review]

**Pull request merged**